src/site/content/en/blog/webcodecs/index.md

```diff
@@ -50,78 +50,92 @@ streaming, etc.
+So it can be used in functions like [`drawImage()`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/drawImage) and[`texImage2D()`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texImage2D), also it can be constructed from canvases, bitmaps, video elements and other video frames.
+
+
+WebCodecs API works well in tandem with the classes from [Insertable Streams API](https://w3c.github.io/mediacapture-transform/)
+which connect WebCodecs to [media stream tracks](https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamTrack).
+- `MediaStreamTrackProcessor` breaks up into media tracks into individual frames
+- `MediaStreamTrackGenerator` helps to create a media track from a stream of frames.
+
 
-Currently in WebCodecs the only way to show a frame on the page is to convert it
-into an
-[`ImageBitmap`](https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap)
-and either draw the bitmap on a canvas or convert it into a
-[`WebGLTexture`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLTexture).
+## WebCodecs and web workers
+
+By design WebCodecs API does all the heavy lifting asynchronously and off the main thread,
+but since frame and chunk callbacks can often be called multiple times a second,
+they might clutter the main thread and thus make the website less responsive.
+Therefore it is preferable to move handling of individual frames and encoded chunks into a
+web worker.
+
+To help with that, [ReadableStream](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream)
+provides a convenient way to automatically transfer all frames coming from a media
+track to the worker. For example, `MediaStreamTrackProcessor` can be used to obtain a
+`ReadableStream` for a media stream track coming from the web camera, after that
+the stream is transferred to a web worker where frames are read one by one and queued
+into a `VideoEncoder`.
+
+With [`HTMLCanvasElement.transferControlToOffscreen`](https://developers.google.com/web/updates/2018/08/offscreen-canvas#unblock_main_thread) even rendering can be done off the main thread. But if all the hight level tools turned
+out to be inconvenient, `VideoFrame` itself is [transferable](https://developer.mozilla.org/en-US/docs/Web/API/Transferable) and may be
+moved between workers.
 
 ## WebCodecs in action
 
 ### Encoding
 
-It all starts with a `VideoFrame`. There are two ways to convert existing
-pictures into `VideoFrame` objects.
-
-The first is to create a frame directly from an
-[`ImageBitmap`](https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap).
-Just call the `VideoFrame()` constructor and give it a bitmap and a presentation
-timestamp.
+```html
+  <Img src="Encoding.png" alt="The path from a Canvas or an ImageBitmap to the network or to storage" width="960" height="720" />
```

Comment:
```suggestion
  <img src="Encoding.png" alt="The path from a Canvas or an ImageBitmap to the network or to storage" width="960" height="720" />
```

Reply:
Is this an example you're working with in the following section? I don't see how it relates since it's not a video.

Reply:
I just marked an image that needs to be used here as an illustration.

This kinda stuff didn't work for me, because I don't know how to upload imaged to the CDN
<figure class="w-figure">
  {% Img src="image/admin/7LNdjNlwCUB1csF9DDl8.png", alt="The path from ImageBitmap to the network or to storage.", width="800", height="291" %}
  <figcaption class="w-figcaption">The path from <code>ImageBitmap</code> to the network or to storage.</figcaption>
</figure>